admin/provisioner/kubernetes.go

```diff
@@ -260,7 +261,11 @@ func (p *KubernetesProvisioner) AwaitReady(ctx context.Context, provisionID stri
 	retryClient.RetryWaitMin = 1 * time.Second
 	retryClient.RetryWaitMax = 3 * time.Second
 	retryClient.Logger = nil // Disable inbuilt logger
-	_, err = retryClient.Get(p.getHost(provisionID) + "/v1/ping")
+	pingURL, err := url.JoinPath(p.getHost(provisionID), "/v1/ping")
+	if err != nil {
+		return err
+	}
+	_, err = retryClient.Get(pingURL)
```

Comment:
If you follow the docs through to the underlying `net/http` call, I think you might find that it is required to get the response and call `.Close()` on the response body. See docstring here: https://pkg.go.dev/net/http#Get

Reply:
Added - good catch